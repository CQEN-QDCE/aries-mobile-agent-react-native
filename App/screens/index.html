<html>
    <head>
        <title>Page Title</title>
    </head>
    <body>
        <script>
            window.hello = function () {
                alert('Bonjour!&&&&&&&&&&&&&&&&'); 
            };
            window.addEventListener('message',function(event){
                alert("Message received from RN: ",event.data)
            },false);
        </script>
        <h1>LÃ  c,est le bon</h1>
        <p>My first paragraph.</p>

    </body>
</html>
