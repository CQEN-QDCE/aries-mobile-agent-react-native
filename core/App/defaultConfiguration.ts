import defaultIndyLedgers from '../configs/ledgers/indy'

import Record from './components/record/Record'
import HomeContentView from './components/views/HomeContentView'
import { minPINLength } from './constants'
import { ConfigurationContext } from './contexts/configuration'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'
import * as oca from './types/oca'
import { PinSecurityLevel } from './types/security'

export const defaultConfiguration: ConfigurationContext = {
  pages: OnboardingPages,
  splash: Splash,
  terms: Terms,
  homeContentView: HomeContentView,
  OCABundle: new oca.DefaultOCABundleResolver().loadDefaultBundles(),
  useBiometry: UseBiometry,
  record: Record,
  pinSecurity: { level: PinSecurityLevel.Level3, minLength: minPINLength, maxLength: minPINLength },
  indyLedgers: defaultIndyLedgers,
  settings: [],
}
