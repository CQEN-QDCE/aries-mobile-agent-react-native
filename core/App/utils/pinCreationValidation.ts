import { minPINLength } from '../../lib/module/constants'
import { PinSecurityLevel } from '../types/security'

const sameNumberTwoTimesConsecutive = new RegExp(/(\d)\1{1,}/) // 2 or more consecutive digits ✔️
const sameNumberThreeTimesConsecutive = new RegExp(/(\d)\1{2,}/) // 3 or more consecutive digits ✔️
const consecutiveSeriesOfThree = new RegExp(/012|123|234|345|456|567|678|789|987|876|765|654|543|432|321|210/) // 3 or more consecutive digits ✔️
const evenNumberSeries = new RegExp('(13579)') // ✔️
const OddNumberSeries = new RegExp('(02468)') // ✔️
const consecutiveTwoNumberRepetition = new RegExp(/([0-9][0-9])\1{1,}/) // ex: 1515 ✔️
const isNumber = new RegExp('^[0-9]+$') // ✔️
const crossNumberPattern = ['159753', '159357', '951357', '951753', '357159', '357951', '753159', '753951'] // ✔️

export interface PinValidationsType {
  isInvalid: boolean
  errorName:
    | 'CrossPatternValidation'
    | 'OddOrEvenSequenceValidation'
    | 'SameNumberTwoTimesConsecutiveValidation'
    | 'ConsecutiveTwoNumbersRepetitionValidation'
    | 'SameNumberThreeTimesConsecutiveValidation'
    | 'SeriesOfThreeNumbersValidation'
    | 'PinOnlyContain6NumbersValidation'
}

export const pinCreationValidations = (pin: string, pinSecurityLevel: number) => {
  const pinValidations: PinValidationsType[] = []
  if (pinSecurityLevel >= PinSecurityLevel.Level6) {
    pinValidations.push({
      isInvalid: crossNumberPattern.includes(pin),
      errorName: 'CrossPatternValidation',
    } as PinValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level5) {
    pinValidations.push({
      isInvalid: evenNumberSeries.test(pin) || OddNumberSeries.test(pin),
      errorName: 'OddOrEvenSequenceValidation',
    } as PinValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level4) {
    pinValidations.push({
      isInvalid: sameNumberTwoTimesConsecutive.test(pin),
      errorName: 'SameNumberTwoTimesConsecutiveValidation',
    } as PinValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level3) {
    pinValidations.push({
      isInvalid: consecutiveTwoNumberRepetition.test(pin),
      errorName: 'ConsecutiveTwoNumbersRepetitionValidation',
    } as PinValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level2) {
    pinValidations.push(
      {
        isInvalid: sameNumberThreeTimesConsecutive.test(pin),
        errorName: 'SameNumberThreeTimesConsecutiveValidation',
      } as PinValidationsType,
      {
        isInvalid: consecutiveSeriesOfThree.test(pin),
        errorName: 'SeriesOfThreeNumbersValidation',
      } as PinValidationsType
    )
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level1) {
    pinValidations.push({
      isInvalid: !isNumber.test(pin) || pin.length !== minPINLength,
      errorName: 'PinOnlyContain6NumbersValidation',
    } as PinValidationsType)
  }
  return pinValidations
}
