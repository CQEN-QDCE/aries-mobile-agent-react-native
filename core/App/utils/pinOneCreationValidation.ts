import { minPINLength } from '../../lib/module/constants'
import { PinSecurityLevel } from '../types/security'

const consecutiveSameNumbersTwoTimes = new RegExp(/(\d)\1{1,}/) // 2 or more consecutive digits ✔️
const consecutiveSameNumbersThreeTimes = new RegExp(/(\d)\1{2,}/) // 3 or more consecutive digits ✔️
const consecutiveSeriesOfThree = new RegExp(/012|123|234|345|456|567|678|789|987|876|765|654|543|432|321|210/) // 3 or more consecutive digits ✔️
const evenNumberSeries = new RegExp('(13579)') // ✔️
const OddNumberSeries = new RegExp('(02468)') // ✔️
const consecutiveTwoNumberRepetition = new RegExp('([0-9]*[0-9])\\1+') // ex: 1515 ✔️
const isNumber = new RegExp('^[0-9]+$') // ✔️
const crossNumberPattern = ['159753', '159357', '951357', '951753', '357159', '357951', '753159', '753951'] // ✔️

export interface PinOneValidationsType {
  isInvalid: boolean
  errorName:
    | 'CrossPatternValidation'
    | 'OddOrEvenSequenceValidation'
    | 'ConsecutiveSameNumbersTwoTimesValidation'
    | 'ConsecutiveTwoNumberRepetitionValidation'
    | 'ConsecutiveSameNumbersThreeTimesValidation'
    | 'SeriesOfThreeNumbersValidation'
    | 'PinOnlyContain6NumbersValidation'
}

export const pinOneCreationValidations = (pinX: string, pinSecurityLevel: number) => {
  const pinOneValidations: PinOneValidationsType[] = []
  if (pinSecurityLevel >= PinSecurityLevel.Level6) {
    pinOneValidations.push({
      isInvalid: crossNumberPattern.includes(pinX),
      errorName: 'CrossPatternValidation',
    } as PinOneValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level5) {
    pinOneValidations.push({
      isInvalid: evenNumberSeries.test(pinX) || OddNumberSeries.test(pinX),
      errorName: 'OddOrEvenSequenceValidation',
    } as PinOneValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level4) {
    pinOneValidations.push({
      isInvalid: consecutiveSameNumbersTwoTimes.test(pinX),
      errorName: 'ConsecutiveSameNumbersTwoTimesValidation',
    } as PinOneValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level3) {
    pinOneValidations.push({
      isInvalid: consecutiveTwoNumberRepetition.test(pinX),
      errorName: 'ConsecutiveTwoNumberRepetitionValidation',
    } as PinOneValidationsType)
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level2) {
    pinOneValidations.push(
      {
        isInvalid: consecutiveSameNumbersThreeTimes.test(pinX),
        errorName: 'ConsecutiveSameNumbersThreeTimesValidation',
      } as PinOneValidationsType,
      {
        isInvalid: consecutiveSeriesOfThree.test(pinX),
        errorName: 'SeriesOfThreeNumbersValidation',
      } as PinOneValidationsType
    )
  }
  if (pinSecurityLevel >= PinSecurityLevel.Level1) {
    pinOneValidations.push({
      isInvalid: !isNumber.test(pinX) || pinX.length !== minPINLength,
      errorName: 'PinOnlyContain6NumbersValidation',
    } as PinOneValidationsType)
  }
  return pinOneValidations
}
