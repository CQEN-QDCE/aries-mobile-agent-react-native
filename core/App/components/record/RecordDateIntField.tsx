import moment from 'moment'
import React, { useEffect, useState } from 'react'
import { StyleSheet, Text, View } from 'react-native'

import { useTheme } from '../../contexts/theme'
import { Attribute, Field } from '../../types/record'
import { testIdWithKey } from '../../utils/testable'

interface RecordBinaryFieldProps {
  field: Field
  shown?: boolean
}
const dateIntFormat = 'YYYYMMDD'

const RecordDateIntField: React.FC<RecordBinaryFieldProps> = ({ field, shown }) => {
  const { ListItems } = useTheme()
  const [date, setDate] = useState<string | undefined>()

  useEffect(() => {
    if (moment((field as Attribute).value as string, dateIntFormat).isValid()) {
      const dateObject = moment((field as Attribute).value as string, dateIntFormat)
      setDate(moment(dateObject).format(field.format))
    }
    setDate((field as Attribute).value as string)
  })

  const styles = StyleSheet.create({
    text: {
      ...ListItems.recordAttributeText,
    },
    image: {
      height: 150,
      aspectRatio: 1,
      resizeMode: 'contain',
      borderRadius: 10,
    },
  })

  return (
    <Text style={styles.text} testID={testIdWithKey('AttributeValue')}>
      {shown ? date : Array(10).fill('\u2022').join('')}
    </Text>
  )
}

export default RecordDateIntField
