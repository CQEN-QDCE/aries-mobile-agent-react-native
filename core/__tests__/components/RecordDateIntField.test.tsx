import { render } from '@testing-library/react-native'
import React from 'react'

import RecordDateIntField from '../../App/components/record/RecordDateIntField'
import { lengthOfhiddenAttributes } from '../../App/constants'
import { testIdWithKey } from '../../App/utils/testable'

describe('Record DateInt Field Component', () => {
  it('Invalid dateInt render field value as is', async () => {
    const field = {
      name: 'Test',
      format: 'YYYY-MM-DD',
      type: 'DateInt',
      value: 'invalid date',
    }
    const tree = render(<RecordDateIntField field={field} shown={true} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
  it('Invalid dateInt render should equal field value', async () => {
    const field = {
      name: 'Test',
      format: 'YYYY-MM-DD',
      type: 'DateInt',
      value: 'invalid date',
    }
    const tree = render(<RecordDateIntField field={field} shown={true} />)
    const fieldValueText = tree.getByTestId(testIdWithKey('AttributeValue'))

    expect(fieldValueText.children[0]).toEqual(field.value)
  })
  it('Valid dateInt render field value in format', async () => {
    const field = {
      name: 'Test',
      format: 'YY-MM-DD',
      type: 'DateInt',
      value: '20000101',
    }
    const tree = render(<RecordDateIntField field={field} shown={true} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
  it('Valid dateInt and date format with only year should render only field value year', async () => {
    const field = {
      name: 'Test',
      format: 'YYYY',
      type: 'DateInt',
      value: '20000101',
    }
    const tree = render(<RecordDateIntField field={field} shown={true} />)
    const fieldValueText = tree.getByTestId(testIdWithKey('AttributeValue'))

    expect(fieldValueText.children[0]).toEqual(field.value.substring(0, 4))
  })
  it('Valid dateInt and random date format should render field value in random format', async () => {
    const field = {
      name: 'Test',
      format: 'YYYY re',
      type: 'DateInt',
      value: '20000101',
    }
    const tree = render(<RecordDateIntField field={field} shown={true} />)
    const fieldValueText = tree.getByTestId(testIdWithKey('AttributeValue'))

    expect(fieldValueText.children[0]).toEqual(field.value.substring(0, 4) + ' re')
  })
  it('Hidden field', async () => {
    const field = {
      name: 'Test',
      format: 'YY-MM-DD',
      type: 'DateInt',
      value: '20000101',
    }
    const tree = render(<RecordDateIntField field={field} shown={false} />)

    const hiddenFieldText = tree.getByTestId(testIdWithKey('AttributeValue'))

    expect(hiddenFieldText.children[0]).toEqual(Array(lengthOfhiddenAttributes).fill('\u2022').join(''))
  })
})
