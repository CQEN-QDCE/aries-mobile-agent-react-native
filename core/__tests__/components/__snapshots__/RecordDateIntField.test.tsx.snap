// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record DateInt Field Component Invalid dateInt render field value as is 1`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 18,
      "fontWeight": "normal",
    }
  }
  testID="com.ariesbifold:id/AttributeValue"
>
  invalid date
</Text>
`;

exports[`Record DateInt Field Component Valid dateInt render field value in format 1`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 18,
      "fontWeight": "normal",
    }
  }
  testID="com.ariesbifold:id/AttributeValue"
>
  00-01-01
</Text>
`;
